cache/blobs.go

```diff
@@ -111,12 +112,21 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 				if release != nil {
 					defer release()
 				}
-				descr, err = sr.cm.Differ.Compare(ctx, lower, upper,
-					diff.WithMediaType(mediaType),
-					diff.WithReference(sr.ID()),
-				)
-				if err != nil {
-					return nil, err
+				if !isTypeWindows(sr) {
+					// Try optimized diff for overlayfs
+					descr, err = sr.computeOverlayBlob(ctx, lower, upper, mediaType, sr.ID())
+					if err != nil {
+						logrus.Errorf("failed to compute blob (%s) from diff of overlay snapshotter: %+v", sr.ID(), err)
```

Comment:
I guess if the native snapshotter is being used this would result in a lot of error logs, which might be a red herring for users. Maybe this could just be Debugf instead?  Or you could leave it as Errorf but have special error type indicating that the snapshotter just wasn't overlay-based and skip logging in that case (that might be unnecessarily complicated though, not sure).

Reply:
Thank you for the review. Fixed to use `Debugf` instead.